File: LatinReader/core/src/main/java/com/ericmschmidt/classicsreader/layouts/TOCListViewAdapter.java
package com.ericmschmidt.classicsreader.layouts;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.ericmschmidt.classicsreader.R;
import com.ericmschmidt.classicsreader.datamodel.TOCEntry;

/**Subclass of ArrayAdapter.
 * Used for displaying chapters in the TOCFragment
 *
 * Layout files:
 * - res/layout/toc_item.xml
 *
 * @author Eric Schmidt
 * @author http://telpirion.com
 * @version 1.5
 * @since 1.4
 */
public class TOCListViewAdapter extends ArrayAdapter<TOCEntry> {
    protected final Context context;
    protected final TOCEntry[] values;

    public TOCListViewAdapter(Context context, TOCEntry[] values) {
        super(context, -1, values);

        this.context = context;
        this.values = values;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View rowView = inflater.inflate(R.layout.toc_listviewitem, parent, false);

        TextView entry = (TextView) rowView.findViewById(R.id.tocitem_entry);
        entry.setText(values[position].toString());

        return rowView;
    }
}

File: LatinReader/views/src/main/java/com/ericmschmidt/classicsreader/ui/layouts/TOCListViewAdapter.kt
package com.ericmschmidt.classicsreader.ui.layouts

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import com.ericmschmidt.classicsreader.views.databinding.TocListviewitemBinding
import com.ericmschmidt.classicsreader.datamodel.TOCEntry

/**
 * Subclass of ArrayAdapter, used for displaying chapters in the TOCFragment.
 *
 * Layout files:
 * - res/layout/toc_item.xml
 *
 * @author Eric Schmidt
 * @author <a href="http://telpirion.com">...</a>
 * @version 2.0
 * @since 1.4
 */
class TOCListViewAdapter(context: Context, private val values: Array<TOCEntry>) :
    ArrayAdapter<TOCEntry>(context, -1, values) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val binding = if (convertView == null) {
            val inflater = LayoutInflater.from(context)
            TocListviewitemBinding.inflate(inflater, parent, false)
        } else {
            TocListviewitemBinding.bind(convertView)
        }

        binding.tocitemEntry.text = values[position].toString()

        return binding.root
    }
}
